src/components/ProteinStructureView.vue

```diff
@@ -62,39 +89,81 @@ export default {
-        return {
-          id: $element.attr('id'),
-          method: $element.find('property[type="method"]').first().attr('value'),
-          resolution: $element.find('property[type="resolution"]').first().attr('value'),
-          chains: $element.find('property[type="chains"]').first().attr('value')
+    selectionDataWithSelectedColorBy: function() {
+        return _.map(this.selectionData, (x) => ({
+          start_residue_number: x.start_residue_number,
+          end_residue_number: x.end_residue_number,
+          color: _.get(x, this.colorBy, '#000')
+        }))
+    },
+    alphaFoldData: function() {
+        if (!this.uniprotData) {
+          return []
         }
-      }).get().filter((pdb) => pdb.id != null)
-    }
+        return $('entry dbReference[type="AlphaFoldDB"]', this.uniprotData).map((i, element) => {
+          const $element = $(element)
+          return {
+            id: $element.attr('id'),
+            method: $element.find('property[type="method"]').first().attr('value'),
+            resolution: $element.find('property[type="resolution"]').first().attr('value'),
+            chains: $element.find('property[type="chains"]').first().attr('value')
+          }
+        }).get().filter((x) => x.id != null)
+    },
   },
 
   mounted: function() {
     this.render()
   },
 
+  setup(props) {
+    const colorBy = ref('mean.color')
+
+    watch(() => props.rowSelected, (newValue) => {
+      if (_.isNumber(newValue)) {
+        colorBy.value = [newValue, 'color']
+      } else {
+        colorBy.value = 'mean.color'
+      }
+    })
+
+    return {
+      colorBy,
+    }
+  },
+
   watch: {
-    pdbs: {
+    colorBy: {
+      handler: function() {
+        this.viewerInstance.visual.select({data: this.selectionDataWithSelectedColorBy})
+      },
+    },
+    selectedResidueRanges: {
+      handler: function(newValue) {
+        const selectedRanges = newValue.map((x) => ({
+          start_residue_number: x.start,
+          end_residue_number: x.end,
+          color: null,
+          focus: true
+        }))
+        this.viewerInstance.visual.select({data:[...this.selectionDataWithSelectedColorBy, ...selectedRanges]})
+        this.viewerInstance.visual.highlight({
+            data: selectedRanges,
+        })
+      },
+      deep: true,
+    },
+    alphaFoldData: {
       handler: function() {
-        let newSelectedPdb = null
-        if (this.selectedPdb) {
-          newSelectedPdb = this.pdbs.find((pdb) => pdb.id == newSelectedPdb.id)
+        let newSelectedAlphaFold = null
+        if (this.selectedAlphaFold) {
+          newSelectedAlphaFold = this.alphaFoldData.find((x) => x.id == newSelectedAlphaFold.id)
         }
-        if (!this.selectedPdb && this.pdbs.length > 0) {
-          this.selectedPdb = this.pdbs[0]
+        if (!this.selectedAlphaFold && this.alphaFoldData.length > 0) {
+          this.selectedAlphaFold = this.alphaFoldData[0]
         }
       }
     },
```

Comment:
Might not be understanding this fully, but could this be more simply written as the following?

```suggestion
     handler: function() {
        if (this.selectedAlphaFold) {
          this.selectedAlphaFold = this.alphaFoldData.find((x) => x.id == this.selectedAlphaFold.id)
       }
        if (!this.selectedAlphaFold && this.alphaFoldData.length > 0) {
          this.selectedAlphaFold = this.alphaFoldData[0]
       }
     }
   },
```

It seemed like we didn't need that temporary newSelectedAlphaFold variable / it wasn't doing anything.